### PR TITLE
[#FLAT-457] Add pagination to newsletters archive

### PR DESCRIPTION
**Pull request by ArthurZaharov**
https://sourceclear.atlassian.net/browse/FLAT-457

### CODE REVIEW

File: app/controllers/discourse_reports/newsletters_controller.rb [hunk 1]
```diff
@@ -11,11 +11,29 @@ def download
...
-      render json: serialize_data(Newsletter.all, NewsletterArchiveSerializer), root: "newsletters"
+      render_json_dump(newsletters: serialize_data(newsletters, NewsletterArchiveSerializer),
+                       latest_newsletter: serialize_data(latest_newsletter, NewsletterArchiveSerializer, root: false),
+                       total_pages: total_pages)
     end
 
     private
 
+    def page
+      (params[:page] || 0).to_i
```
Comment (ramil350) [inline app/controllers/discourse_reports/newsletters_controller.rb]: Maybe just

``` ruby
params[:page].to_i
```

File: app/controllers/discourse_reports/newsletters_controller.rb [hunk 2]
```diff
@@ -11,11 +11,29 @@ def download
...
+    end
+
+    def total_pages
+      Newsletter.all.count / 10
```
Comment (ramil350) [inline app/controllers/discourse_reports/newsletters_controller.rb:26]: Do we need all?

``` ruby
Newsletter.count
```

File: app/controllers/discourse_reports/newsletters_controller.rb [hunk 3]
```diff
@@ -11,11 +11,29 @@ def download
...
+    end
+
+    def newsletters
+      Newsletter.all.offset(page * 10).limit(10)
+    end
+
+    def latest_newsletter
+      Newsletter.all.first
```
Comment (RamilGilmanov) [inline app/controllers/discourse_reports/newsletters_controller.rb]: so is here

File: app/controllers/discourse_reports/newsletters_controller.rb (hunk 2 above)
Comment (ramil350) [inline app/controllers/discourse_reports/newsletters_controller.rb:26]: Oh, it's not an AR model, sorry

**Comment by ramil350** [comment]
:+1:

**Pull request merged**